projects/isy-angular-widgets/src/lib/incomplete-date/incomplete-date.service.ts

```diff
@@ -70,8 +70,8 @@ export class IncompleteDateService {
 
     if (
       dateStr.match(INCOMPLETE_DATE_REGEX) !== null ||
-      dateObject.month.includes(this.DATE_CHAR) ||
-      dateObject.year.includes(this.DATE_CHAR)
+      dateObject.month!.includes(this.DATE_CHAR) ||
+      dateObject.year!.includes(this.DATE_CHAR)
```

Comment:
this seems questionable. if we are sure that month and year are present, why are they marked as optional in the interface?

Reply:
That was a part of a very difficult ESLint fix.
Are we 100% sure that month and year are always present?

If yes, we must revert the code changes.

Reply:
That doesn't explain why the month and year are defined optional in the first place.

Reply:
You are just describing how you tricked ESLint to shut up, but I don't see a solution to WHY it complained in the first place.

All you did is to mark a property optional and then at runtime assuming (! NOT ENSURING !) that it is there regardless.

Reply:
Can you suggest a better solution?

If not we have 3 options:
1. We don't do anything
2. I revert the code and disable ESLint for line 59
3. I search for a new solution, change functional code (that is not part of a migration) and block the Angular 16 Migration for another day

Reply:
Changing functional code is indeed part of a migration.
Reducing code quality and introducing the need for refactoring just to get a compiling artifact to save on time meanwhile is not; sometimes it is necessary slopiness, but it is not here.